router/core/flushwriter.go

```diff
@@ -109,6 +111,9 @@ func (f *HttpFlushWriter) Flush() (err error) {
 	if err != nil {
 		return err
 	}
+
+	f.mu.Lock()
```

Comment:
@df-wg Did you figure out why this is necessary or is there even an alternative approach?